projects/core/src/lib/colors/color-palette-editor/color-palette-editor.component.ts

```diff
@@ -0,0 +1,132 @@
+    colorMap = new Map<number, Color>(); // Returned to the parent component as parameter for the colorizer
+
+    constructor(private ref: ChangeDetectorRef) {}
+
+    ngOnInit(): void {
+        this.initColorInputs();
+    }
+
+    /**
+     * (re)creates the ColorMap as well as the ColorAttributeInputs based on the current layer
+     */
+    initColorInputs(): void {
+        this.colorMap = new Map<number, Color>();
+        this.allColors = new Array<ColorAttributeInput>();
+
+        this.getPaletteColorizer()
+            .getBreakpoints()
+            .forEach((bp, index) => {
+                const newInput = {key: bp.value.toString(), value: bp.color};
+                this.allColors.push(newInput);
+                this.colorMap.set(bp.value, this.getPaletteColorizer().getColorAtIndex(index));
+            });
+    }
+
+    setSymbology(newSymbology: RasterSymbology): void {
+        this.symbology = newSymbology;
+    }
+
+    getColors(): Map<number, Color> {
+        return this.colorMap;
+    }
+
+    getPaletteColorizer(): PaletteColorizer {
+        return this.symbology.colorizer as PaletteColorizer;
+    }
+
+    createColorAttributeInputs(): ColorAttributeInput[] {
+        const inputs: ColorAttributeInput[] = [];
+        for (let i = 0; i < this.getPaletteColorizer().getNumberOfColors(); i++) {
+            const newColor: ColorAttributeInput = {key: i.toString(), value: this.getPaletteColorizer().getColorAtIndex(i)};
+            inputs.push(newColor);
+            this.colorMap.set(i, newColor.value);
+        }
+        return inputs;
+    }
+
+    // Currently not used ... everything is done by rebuildColorMap, can be deleted...
+    updateColor(pos: number, color: ColorAttributeInput): void {
+        this.allColors[pos] = color; // because else, changes to HTML won't affect allColors
+        this.colorMap.set(parseFloat(color.key), color.value); // Sets the key/value in the map according to the ColorAttributeInput. Old map value still exists after this!
+
+        this.emitSymbology();
+    }
+
+    /**
+     * Recreate the color map so that only values for which a ColorAttributeInput exists
+     * are contained within the map. This is necessary because the $event that gets
+     * passed to updateColor won't contain the previous rasterValue to delete in manually at
+     * the time of updating.
+     */
+    rebuildColorMap(pos: number, color: ColorAttributeInput): void {
+        this.allColors[pos] = color; // need this, or changes to HTML won't affect allColors
+
+        // Recreate the map
+        this.colorMap = new Map<number, Color>();
+        this.allColors.forEach((input: ColorAttributeInput) => {
+            const colorKey: number = parseFloat(input.key);
+            const colorValue: Color = input.value;
+            this.colorMap.set(colorKey, colorValue);
+        });
+        this.emitSymbology();
+    }
+
```

Comment:
Please pull the current master, since the histogram def has changed and it now doesn't work with the current master backend (for testing).